Dart/src/com/jetbrains/lang/dart/analyzer/DartAnalysisServerService.java

```diff
@@ -2244,6 +2314,23 @@ private String getShortErrorMessage(@NotNull String methodName, @Nullable String
            ", error code = " + error.getCode() + ": " + error.getMessage();
   }
 
+  private void logTookTooLongMessage(@NotNull final String methodName,
+                                     final long timeout,
+                                     @Nullable final String contextID,
+                                     @Nullable String filePath) {
+    StringBuilder builder = new StringBuilder();
+    builder.append(methodName).append("() took longer than ").append(timeout).append("ms");
+    if (contextID != null) {
+      builder.append(", for context id ").append(contextID);
+    }
+    if (filePath != null) {
+      builder.append(", for file ").append(filePath);
+    }
+    builder.append(", IntelliJ product name: ").append(getClientId()).append(", ").append(getClientVersion());
```

Comment:
IDE version is always in logs. I'll remove this line if you don't mind.

Reply:
SGTM